sw/device/silicon_creator/lib/drivers/flash_ctrl.c

```diff
@@ -0,0 +1,100 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/drivers/flash_ctrl.h"
+
+#include "sw/device/silicon_creator/lib/error.h"
+
+#include "flash_ctrl_regs.h"  // Generated.
+
+static bool is_busy(const flash_ctrl_t *flash_ctrl) {
+  return mmio_region_get_bit32(flash_ctrl->base_addr,
+                               FLASH_CTRL_CTRL_REGWEN_REG_OFFSET,
+                               FLASH_CTRL_CTRL_REGWEN_EN_BIT) == 0;
+}
+
+void flash_ctrl_init(const flash_ctrl_t *flash_ctrl) {
+  // Initialize the flash controller.
+  mmio_region_write32(flash_ctrl->base_addr, FLASH_CTRL_INIT_REG_OFFSET,
+                      bitfield_bit32_write(0u, FLASH_CTRL_INIT_VAL_BIT, true));
+}
+
+flash_ctrl_status_t flash_ctrl_status_get(const flash_ctrl_t *flash_ctrl);
```

Comment:
Does this have an implementation?  I can't seem to find it.

Reply:
@jon-flatley PTAL